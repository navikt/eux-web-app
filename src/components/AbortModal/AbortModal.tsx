import { FlexDiv } from 'components/StyledComponents'
import Modal from 'nav-frontend-modal'
import { Normaltekst, Undertittel } from 'nav-frontend-typografi'
import NavHighContrast, { HighContrastHovedknapp, HighContrastKnapp } from 'nav-hoykontrast'
import PT from 'prop-types'
import React from 'react'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components'

if (document && document.getElementById('root')) {
  Modal.setAppElement('#root')
}

const Content = styled.div`
  margin: 1.5em 2em 3em 2em;
  text-align: center;
`

const Text = styled(Normaltekst)`
  padding: 2em;
  margin-bottom: 1em;
`
export interface AbortModalProps {
  closeModal: () => void
  highContrast: boolean
  isOpen?: boolean
  onAbort: () => void
}

const AbortModal: React.FC<AbortModalProps> = ({
  closeModal, highContrast, isOpen = false, onAbort
}: AbortModalProps): JSX.Element => {
  const { t } = useTranslation()
  return (
    <Modal
      isOpen={isOpen}
      onRequestClose={closeModal}
      closeButton={false}
      contentLabel={t('app:modal-abort-contentLabel')}
    >
      <NavHighContrast highContrast={highContrast}>
        <Content>
          <Undertittel>
            {t('app:modal-abort-title')}
          </Undertittel>
          <Text>
            {t('app:modal-abort-description')}
          </Text>
          <FlexDiv>
            <HighContrastHovedknapp onClick={onAbort}>
              {t('app:modal-abort-yes-cancel')}
            </HighContrastHovedknapp>
            <HighContrastKnapp onClick={closeModal}>
              {t('app:modal-abort-no-continue')}
            </HighContrastKnapp>
          </FlexDiv>
        </Content>
      </NavHighContrast>
    </Modal>
  )
}

AbortModal.propTypes = {
  closeModal: PT.func.isRequired,
  highContrast: PT.bool.isRequired,
  isOpen: PT.bool,
  onAbort: PT.func.isRequired
}

export default AbortModal
