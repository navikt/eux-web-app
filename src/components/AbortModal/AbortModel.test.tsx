
import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import AbortModal, { AbortModalProps } from './AbortModal'

describe('components/AbortModal/AbortModal', () => {
  let wrapper: ReactWrapper
  const initialMockProps: AbortModalProps = {
    closeModal: jest.fn(),
    onAbort: jest.fn(),
    isOpen: true
  }

  it('Renders', () => {
    wrapper = mount(<AbortModal {...initialMockProps} />)
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('HTML structure', () => {
    wrapper = mount(<AbortModal {...initialMockProps} />)
    expect(wrapper.exists('div.modal__innhold')).toBeTruthy()
  })
})
