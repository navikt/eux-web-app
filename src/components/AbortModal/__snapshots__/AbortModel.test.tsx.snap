// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AbortModal/AbortModal Render: match snapshot 1`] = `
.c0 {
  margin: 1.5em 2em 3em 2em;
  text-align: center;
}

.c1 {
  padding: 2em;
  margin-bottom: 1em;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<AbortModal
  closeModal={[MockFunction]}
  isOpen={true}
  onAbort={[MockFunction]}
>
  <ModalWrapper
    closeButton={false}
    closeTimeoutMS={0}
    contentLabel="ui:modal-abort-contentLabel"
    isOpen={true}
    onRequestClose={[MockFunction]}
    shouldCloseOnOverlayClick={true}
  >
    <Modal
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="modal"
      closeTimeoutMS={0}
      contentLabel="ui:modal-abort-contentLabel"
      isOpen={true}
      onRequestClose={[Function]}
      overlayClassName="modal__overlay"
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
    >
      <Portal
        containerInfo={
          .c0 {
  margin: 1.5em 2em 3em 2em;
  text-align: center;
}

.c1 {
  padding: 2em;
  margin-bottom: 1em;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
            class="ReactModalPortal"
          >
            <div
              class="ReactModal__Overlay ReactModal__Overlay--after-open modal__overlay"
            >
              <div
                aria-label="ui:modal-abort-contentLabel"
                class="ReactModal__Content ReactModal__Content--after-open modal"
                role="dialog"
                tabindex="-1"
              >
                <section>
                  <div
                    class="c0"
                  >
                    <h2
                      class="typo-undertittel"
                    >
                      ui:modal-abort-title
                    </h2>
                    <p
                      class="typo-normal c1"
                    >
                      ui:modal-abort-description
                    </p>
                    <div
                      class="c2"
                    >
                      <button
                        class="knapp knapp--hoved"
                        type="submit"
                      >
                        ui:modal-abort-yes-button
                      </button>
                      <button
                        class="knapp knapp--standard"
                        type="submit"
                      >
                        ui:modal-abort-no-continue
                      </button>
                    </div>
                  </div>
                </section>
              </div>
            </div>
          </div>
        }
      >
        <ModalPortal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          className="modal"
          closeTimeoutMS={0}
          contentLabel="ui:modal-abort-contentLabel"
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={true}
          onRequestClose={[Function]}
          overlayClassName="modal__overlay"
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          role="dialog"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {},
              "overlay": Object {},
            }
          }
        >
          <div
            className="ReactModal__Overlay ReactModal__Overlay--after-open modal__overlay"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={Object {}}
          >
            <div
              aria-label="ui:modal-abort-contentLabel"
              className="ReactModal__Content ReactModal__Content--after-open modal"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              role="dialog"
              style={Object {}}
              tabIndex="-1"
            >
              <section>
                <styled.div>
                  <div
                    className="c0"
                  >
                    <Undertittel>
                      <TypografiBase
                        type="undertittel"
                      >
                        <h2
                          className="typo-undertittel"
                        >
                          ui:modal-abort-title
                        </h2>
                      </TypografiBase>
                    </Undertittel>
                    <Styled(Normaltekst)>
                      <Normaltekst
                        className="c1"
                      >
                        <TypografiBase
                          className="c1"
                          type="normaltekst"
                        >
                          <p
                            className="typo-normal c1"
                          >
                            ui:modal-abort-description
                          </p>
                        </TypografiBase>
                      </Normaltekst>
                    </Styled(Normaltekst)>
                    <styled.div>
                      <div
                        className="c2"
                      >
                        <Hovedknapp
                          onClick={[MockFunction]}
                        >
                          <KnappBase
                            autoDisableVedSpinner={false}
                            className={null}
                            disabled={false}
                            htmlType="submit"
                            inaktivKlasseVedDisabled={false}
                            kompakt={false}
                            mini={false}
                            onClick={[MockFunction]}
                            spinner={false}
                            type="hoved"
                          >
                            <button
                              className="knapp knapp--hoved"
                              disabled={false}
                              onClick={[MockFunction]}
                              type="submit"
                            >
                              ui:modal-abort-yes-button
                            </button>
                          </KnappBase>
                        </Hovedknapp>
                        <Knapp
                          onClick={[MockFunction]}
                        >
                          <KnappBase
                            autoDisableVedSpinner={false}
                            className={null}
                            disabled={false}
                            htmlType="submit"
                            inaktivKlasseVedDisabled={false}
                            kompakt={false}
                            mini={false}
                            onClick={[MockFunction]}
                            spinner={false}
                            type="standard"
                          >
                            <button
                              className="knapp knapp--standard"
                              disabled={false}
                              onClick={[MockFunction]}
                              type="submit"
                            >
                              ui:modal-abort-no-continue
                            </button>
                          </KnappBase>
                        </Knapp>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </section>
            </div>
          </div>
        </ModalPortal>
      </Portal>
    </Modal>
  </ModalWrapper>
</AbortModal>
`;
